pkg/controllers/nodepool/registrationhealth/suite_test.go

```diff
@@ -0,0 +1,131 @@
+	It("should ignore setting NodeRegistrationHealthy status condition on NodePools which aren't managed by this instance of Karpenter", func() {
+		nodePool.Spec.Template.Spec.NodeClassRef = &v1.NodeClassReference{
+			Group: "karpenter.test.sh",
+			Kind:  "UnmanagedNodeClass",
+			Name:  "default",
+		}
+		ExpectApplied(ctx, env.Client, nodePool, nodeClass)
+		_ = ExpectObjectReconciled(ctx, env.Client, controller, nodePool)
+		nodePool = ExpectExists(ctx, env.Client, nodePool)
+		Expect(nodePool.StatusConditions().Get(v1.ConditionTypeNodeRegistrationHealthy)).To(BeNil())
+	})
+	It("should not set NodeRegistrationHealthy status condition on nodePool when nodeClass does not exist", func() {
+		ExpectApplied(ctx, env.Client, nodePool)
+		ExpectObjectReconciled(ctx, env.Client, controller, nodePool)
+		nodePool = ExpectExists(ctx, env.Client, nodePool)
+		Expect(nodePool.StatusConditions().Get(v1.ConditionTypeNodeRegistrationHealthy)).To(BeNil())
+	})
+	It("should set NodeRegistrationHealthy status condition on nodePool as Unknown if the nodeClass observed generation doesn't match with that on nodePool", func() {
+		nodePool.StatusConditions().SetFalse(v1.ConditionTypeNodeRegistrationHealthy, "unhealthy", "unhealthy")
+		nodePool.Status.NodeClassObservedGeneration = int64(1)
+		ExpectApplied(ctx, env.Client, nodePool, nodeClass)
+
+		nodePool.Spec.Limits = map[corev1.ResourceName]resource.Quantity{corev1.ResourceCPU: resource.MustParse("14")}
+		ExpectApplied(ctx, env.Client, nodePool, nodeClass)
+		_ = ExpectObjectReconciled(ctx, env.Client, controller, nodePool)
+		nodePool = ExpectExists(ctx, env.Client, nodePool)
+		Expect(nodePool.StatusConditions().Get(v1.ConditionTypeNodeRegistrationHealthy).IsUnknown()).To(BeTrue())
+		Expect(nodePool.Status.NodeClassObservedGeneration).To(Equal(int64(1)))
+	})
+	It("should set NodeRegistrationHealthy status condition on nodePool as Unknown if the nodePool is updated", func() {
+		ExpectApplied(ctx, env.Client, nodePool, nodeClass)
+		_ = ExpectObjectReconciled(ctx, env.Client, controller, nodePool)
+		nodePool = ExpectExists(ctx, env.Client, nodePool)
+		Expect(nodePool.StatusConditions().Get(v1.ConditionTypeNodeRegistrationHealthy).IsUnknown()).To(BeTrue())
+		Expect(nodePool.Status.NodeClassObservedGeneration).To(Equal(int64(1)))
+	})
+	It("should not set NodeRegistrationHealthy status condition on nodePool as Unknown if it is already set to true", func() {
+		nodePool.StatusConditions().SetTrue(v1.ConditionTypeNodeRegistrationHealthy)
+		nodePool.Status.NodeClassObservedGeneration = int64(1)
+		ExpectApplied(ctx, env.Client, nodePool, nodeClass)
+		_ = ExpectObjectReconciled(ctx, env.Client, controller, nodePool)
+		nodePool = ExpectExists(ctx, env.Client, nodePool)
+		Expect(nodePool.StatusConditions().Get(v1.ConditionTypeNodeRegistrationHealthy).IsUnknown()).To(BeFalse())
+	})
+	It("should not set NodeRegistrationHealthy status condition on nodePool as Unknown if it is already set to false", func() {
```

Comment:
I think (at least) one of these tests should validate that we are properly updating the NodeClassObservedGeneration since that's the responsibility of this controller

Reply:
Updated